purchase_invoicing_no_zero_line/tests/test_purchase_invoicing_no_zero_line.py

```diff
@@ -85,20 +85,41 @@ def setUpClass(cls):
     def test_01_all_lines(self):
         self.purchase_order.button_confirm()
         self.purchase_order.order_line[0].qty_received = 5
-        view = self.purchase_order.with_context(create_bill=True).action_view_invoice()
-        context = view["context"]
-        invoice = Form(self.env["account.move"].with_context(context)).save()
+        supplier_form = self.env.ref("account.view_move_form")
+        invoice_form = Form(
+            self.env["account.move"].with_context(default_move_type="in_invoice"),
+            view=supplier_form,
+        )
+        po = self.env["purchase.bill.union"].search(
+            [("purchase_order_id", "=", self.purchase_order.id)]
+        )
+        invoice_form.purchase_vendor_bill_id = po
+        invoice = invoice_form.save()
```

Comment:
This same change is basically repeated below two more times, wouldn't it be nice to have a method doing this so it is not repeated?
Not blocking since this was repeated before this PR already, but nice to have.

Moreover, there are other nice ways to find the purchase order, you can have a look at https://github.com/odoo/odoo/blob/1f951c8542cf8974107ad6710ff4d2f78267a199/addons/purchase/tests/test_purchase_invoice.py#L519

Reply:
:+1: